### PR TITLE
Refactor: Calculate invoice expiry outside of the crud

### PR DESCRIPTION
**Pull request by callebtc**

### CODE REVIEW

**Review by dni — APPROVED** [review]

**Comment by prusnak** [comment]
Resolved the conflicts and rebased on top of the `dev` branch.

File: lnbits/core/crud.py [hunk 1]
```diff
@@ -5,7 +5,6 @@
 from uuid import UUID, uuid4
 
 import shortuuid
-from bolt11.decode import decode
```
Comment (motorina0) [inline lnbits/core/crud.py:8]: - what a blasphemy!

File: lnbits/core/services.py [hunk 2]
```diff
@@ -196,20 +197,24 @@ async def pay_invoice(
...
             memo: str
+            expiry: Optional[datetime.datetime]
             extra: Optional[Dict]
 
         payment_kwargs: PaymentKwargs = PaymentKwargs(
             wallet_id=wallet_id,
             payment_request=payment_request,
             payment_hash=invoice.payment_hash,
             amount=-invoice.amount_msat,
+            expiry=invoice_expiry,
```
Comment (motorina0) [inline lnbits/core/services.py:217]: - nitpick: is a variable for `invoice_expiry` required?
```suggestion
            expiry=get_bolt11_expiry(payment_request),
```

**Review by motorina0 — DISMISSED** [review]

**Review by dni — APPROVED** [review]

**Review by dni — CHANGES_REQUESTED** [review]
screenshot-1696921282

**Review by dni — APPROVED** [review]
1 expiry on create_payment was missing, works now for me

**Review by motorina0 — APPROVED** [review]

**Pull request merged**